components/EmailValidator.vue

```diff
@@ -35,13 +37,15 @@
   </div>
 </template>
 <script>
-import { Field, ErrorMessage } from 'vee-validate'
+import { Field, ErrorMessage, Form as VeeForm } from 'vee-validate'
 import { useDomainStore } from '../stores/domain'
 import { ref } from '#imports'
 import { EMAIL_REGEX } from '~/constants'
 
+const domainValidationCache = new Map()
```

Comment:
Feels overkill, since we are not likely to validate domains more than once before a page refresh.  But since it's there, leave it in.

Reply:
@edwh Well, vee-validate doesn't cache validation results, so that ends up querying Google every 'blur'/'change' event, like 2-3 requests per each "regex-valid" domain. I've thought having a small cache would help :)